src/Analyzers/Core/Analyzers/UseCollectionInitializer/AbstractObjectCreationExpressionAnalyzer.cs

```diff
@@ -24,166 +22,44 @@ internal abstract class AbstractObjectCreationExpressionAnalyzer<
         where TObjectCreationExpressionSyntax : TExpressionSyntax
         where TVariableDeclaratorSyntax : SyntaxNode
     {
-        protected SemanticModel _semanticModel;
-        protected ISyntaxFacts _syntaxFacts;
+        protected UpdateExpressionState<TExpressionSyntax, TStatementSyntax> State;
```

Comment:
extracted a lot of common state and helper logic into this new type.  That way it could be used by "convert object-creation to XXX" *and* the "convert collection-builder to collection".  

I went through a lot of iterations trying to figure out how to share code without crazy inheritance stuff, and this is what i settled on as the cleanest way.